OWML.ModHelper.Input/ModInputHandler.cs

```diff
@@ -0,0 +1,338 @@
+﻿using System;
+using System.Reflection;
+using System.Collections.Generic;
+using OWML.Common;
+using UnityEngine;
+
+namespace OWML.ModHelper.Input
+{
+    public class ModInputHandler : IModInputHandler
+    {
+        private const float Cooldown = 0.05f;
+        private const float TapDuration = 0.1f;
+        private const int MinUsefulKey = 8;
+        private const int MaxUsefulKey = 350;
+        private const int MaxComboLength = 7;
+        private const string XboxPrefix = "xbox_";
+        private const BindingFlags NonPublic = BindingFlags.NonPublic | BindingFlags.Instance;
+
+        internal static ModInputHandler Instance { get; private set; }
+
+        private HashSet<IModInputCombination> _singlePressed = new HashSet<IModInputCombination>();
+        private Dictionary<long, IModInputCombination> _comboRegistry = new Dictionary<long, IModInputCombination>();
+        private HashSet<InputCommand> _gameBindingRegistry = new HashSet<InputCommand>();
+        private float[] _timeout = new float[MaxUsefulKey];
+        private int[] _gameBindingCounter = new int[MaxUsefulKey];
+        private ModInputCombination _lastPressed;
+        private float _lastUpdate;
+        private float _lastSingleUpdate;
+        private readonly IModLogger _logger;
+        private readonly IModConsole _console;
+
+        internal bool ShouldIgnore(KeyCode code)
+        {
+            UpdateCombo();
+            return _lastPressed != null && Time.realtimeSinceStartup - _timeout[(int)code] < Cooldown;
+        }
+
+        public ModInputHandler(IModLogger logger, IModConsole console, IHarmonyHelper patcher)
+        {
+            _console = console;
+            _logger = logger;
+            patcher.AddPrefix<SingleAxisCommand>("ClearBinding", typeof(InputInterceptor), nameof(InputInterceptor.SingleAxisRemovePre));
+            patcher.AddPrefix<DoubleAxisCommand>("ClearBinding", typeof(InputInterceptor), nameof(InputInterceptor.DoubleAxisRemovePre));
+            patcher.AddPostfix<SingleAxisCommand>("Update", typeof(InputInterceptor), nameof(InputInterceptor.SingleAxisUpdatePost));
+            patcher.AddPostfix<DoubleAxisCommand>("Update", typeof(InputInterceptor), nameof(InputInterceptor.DoubleAxisUpdatePost));
+            Instance = this;
+        }
+
+        private void UpdateCombo()
```

Comment:
This method is very long, could you split it into multiple methods?

Reply:
generating hash from currently pressed buttons could be separated very easily

Reply:
Meaning: keep the method but extract the contents into multiple methods

Reply:
done now

Reply:
now it's 2 ~39 lines methods

Reply:
I don't think I can break it down farther though

Reply:
@TAImatem Did you push those changes? Not seeing any new commits

Reply:
these separations take some time ro figure out, I only just finished

Reply:
and currently I am doublechecking

Reply:
managed to split it in 3